Mehdi/Assignments 1/6.py

```diff
@@ -0,0 +1,90 @@
+from functools import wraps
+import time
+
+print('\n######################################## Runtime ############################################')
+
+def timeit(func):
+    @wraps(func)
+    def timeit_wrapper(*args, **kwargs):
+        start_time = time.perf_counter()
+        result = func(*args, **kwargs)
+        end_time = time.perf_counter()
+        total_time = end_time - start_time
+        print(f'Function {func.__name__} Took {total_time:.4f} seconds')
+        return result
+    return timeit_wrapper
+
+def line_numbers(file_path, word_list): # searching in file
+
+    with open(file_path, 'r') as f:
+        results = {word:[] for word in word_list}
+        for num, line in enumerate(f, start=1):
+            for word in word_list:
+                if word in line:
+                    results[word].append(num)
+                    temp = num
+    return temp
+
+class Mesh: # creating mesh matrix
+    def __init__(self, n, d, e, t): #(number of nodes, dimension, number of elements, element group capacity)
+        self.nodal = [[0] * d for _ in range(n)]
+        self.conn = [[0] * t for _ in range(e)]
+
+    def assign_node(self, n, a, value): # assign node coordinates
+        self.nodal[n][a] = value
+
+    def assign_conn(self, e, a, value): # assign elenment number for connections
+        self.conn[e][a] = value
+
+    def retrieve_node(self, n, a): # reading node information
+        return self.nodal[n][a]
+
+    def retrieve_element(self, e, a): # reading element connectivity information
+        return self.conn[e][a]
+    
+
+with open('mesh.dat', 'r') as d: # open mesh file
+    content = d.readlines()

```

Comment:
readlines() is the worst way to read the data from the file... I covered about this in today's office hours